File: src/main/java/org/hwabeag/divesystem/events/InvCloseEvent.java
package org.hwabeag.divesystem.events;

import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryCloseEvent;
import org.bukkit.inventory.ItemStack;
import org.hwabeag.divesystem.config.ConfigManager;
import org.jetbrains.annotations.Nullable;

public class InvCloseEvent implements Listener {

    FileConfiguration DiveSystemConfig = ConfigManager.getConfig("dive-system");
    FileConfiguration PlayerConfig = ConfigManager.getConfig("player");
    @EventHandler
    public void onClose(InventoryCloseEvent e) {
        Player p = (Player) e.getPlayer();
        String name = p.getName();
        if (e.getView().getTitle().equals("물품설정")) {
            if (PlayerConfig.getString(name + ".설정상점") != null) {
                String shopname = PlayerConfig.getString(name + ".설정상점");
                if (DiveSystemConfig.getString("잠수상점." + shopname) != null) {
                    int index = DiveSystemConfig.getInt("잠수상점." + shopname + ".라인");
                    int page = PlayerConfig.getInt(name + ".페이지");
                    int N = 0;
                    while (N <= index * 9 - 1) {
                        if (e.getInventory().getItem(N) != null) {
                            @Nullable ItemStack item = e.getInventory().getItem(N);
                            DiveSystemConfig.set("잠수상점." + shopname + ".물품." + page + "." + N, item);
                            if (DiveSystemConfig.getString("잠수상점." + shopname + ".금액." + page + "." + N) == null) {
                                DiveSystemConfig.set("잠수상점." + shopname + ".금액." + page + "." + N, 0);
                            }
                            ConfigManager.saveConfigs();
                        } else {
                            if (DiveSystemConfig.getItemStack("잠수상점." + shopname + ".물품." + page + "." + N) != null) {
                                DiveSystemConfig.set("잠수상점." + shopname + ".물품." + page + "." + N, null);
                                DiveSystemConfig.set("잠수상점." + shopname + ".금액." + page + "." + N, null);
                                ConfigManager.saveConfigs();
                            }
                        }
                        N += 1;
                    }
                    PlayerConfig.set(name + ".설정상점", "없음");
                    ConfigManager.saveConfigs();
                }
            }
        }
    }
}

File: src/main/kotlin/org/hwabeag/divesystem/events/InvCloseEvent.kt
package org.hwabeag.divesystem.events

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryCloseEvent
import org.hwabeag.divesystem.config.ConfigManager

class InvCloseEvent : Listener {
    @EventHandler
    fun onClose(event: InventoryCloseEvent) {
        val player = event.player
        val name = player.name
        val diveSystemConfig = ConfigManager.getConfig("dive-system")
        val playerConfig = ConfigManager.getConfig("player")

        if (event.view.title != "물품설정") {
            return
        }
        val shopName = playerConfig.getString("$name.설정상점") ?: return
        if (diveSystemConfig.getString("잠수상점.$shopName") == null) {
            return
        }
        val line = diveSystemConfig.getInt("잠수상점.$shopName.라인")
        val page = playerConfig.getInt("$name.페이지")
        var slot = 0
        while (slot <= line * 9 - 1) {
            val item = event.inventory.getItem(slot)
            if (item != null) {
                diveSystemConfig.set("잠수상점.$shopName.물품.$page.$slot", item)
                if (diveSystemConfig.get("잠수상점.$shopName.금액.$page.$slot") == null) {
                    diveSystemConfig.set("잠수상점.$shopName.금액.$page.$slot", 0)
                }
            } else if (diveSystemConfig.getItemStack("잠수상점.$shopName.물품.$page.$slot") != null) {
                diveSystemConfig.set("잠수상점.$shopName.물품.$page.$slot", null)
                diveSystemConfig.set("잠수상점.$shopName.금액.$page.$slot", null)
            }
            slot += 1
        }
        playerConfig.set("$name.설정상점", "없음")
        ConfigManager.saveConfigs()
    }
}
